homeassistant/components/nuki/__init__.py

```diff
@@ -101,7 +108,16 @@ async def async_update_data():
             # Note: asyncio.TimeoutError and aiohttp.ClientError are already
             # handled by the data update coordinator.
             async with async_timeout.timeout(10):
-                await hass.async_add_executor_job(_update_devices, hass, ent_reg, locks + openers)
+                events = await hass.async_add_executor_job(_update_devices, locks + openers)
+
+                for event, device_ids in events.items():
```

Comment:
Please move this block down, out of the try... except block. We only want to wrap the line that can raise inside the try... except block.

Reply:
But that code shouldn't execute if the update fails. Maybe extracting it to a function?

Reply:
We raise if update fails, so the code won't execute if we put it below those blocks.